chain/index/README.MD

```diff
@@ -0,0 +1,188 @@
+   - Restart your Lotus node with the new configuration.
+   - The `ChainIndexer` will begin indexing **real-time chain state changes** immediately.
+
+Once Lotus starts with the `ChainIndexer` enabled, it will begin indexing real-time chain state changes i.e. new incoming tipsets. However, it will not index any historical chain state i.e. any previously existing chain state. To index historical chain state (aka **"backfilling"**), you can use the following tools that we're shipping with the `ChainIndexer`:
+
+### The `ChainValidateIndex` JSON RPC API
+
+The `ChainValidateIndex` JSON RPC API serves a dual purpose: it validates/diagnoses the integrity of the index at a specific epoch (i.e., it ensures consistency between indexed data and actual chain state), while also providing the option to backfill the `ChainIndexer` if it does have data for the specified epoch. 
+
+```go
+// IndexValidation contains detailed information about the validation status of a specific chain epoch.
+type IndexValidation struct {
+	// TipSetKey is the key of the canonical tipset for this epoch.
+	TipSetKey TipSetKey
+	// Height is the epoch height at which the validation is performed.
+	Height uint64
+	// IndexedMessagesCount indicates the number of indexed messages for the canonical tipset at this epoch.
+	IndexedMessagesCount uint64
+	// IndexedEventsCount signifies the number of indexed events for the canonical tipset at this epoch.
+	IndexedEventsCount uint64
+	// Backfilled denotes whether missing data was successfully backfilled during validation.
+	Backfilled bool
+	// IsNullRound indicates if the epoch corresponds to a null round and therefore does not have any indexed messages or events.
+	IsNullRound bool
+}
+
+// ChainValidateIndex validates the integrity of the chain index at a specified epoch and also optionally backfills missing data.
+//
+// Parameters:
+//   - epoch: The specific chain epoch for which to validate/backfill the index.
+//   - backfill: A boolean flag indicating whether to attempt backfilling of missing data if the index does not have data for the 
+//               specified epoch.
+//
+// Returns:
+//   - *types.IndexValidation: A pointer to an IndexValidation struct containing the results of the validation/backfill.
+//   - error: An error object if the validation/backfill fails. The error message will contain details about the index 
+//            corruption if the call fails because of an incosistency between indexed data and the actual chain state.
+//
+// Note: The API returns an error if the index does not have data for the specified epoch and backfill is set to false.
+func ChainValidateIndex(ctx context.Context, epoch abi.ChainEpoch, backfill bool) (*types.IndexValidation, error)
+```
+
+The `ChainValidateIndex` API serves multiple purposes:
+
+1. Validates the chain index at a specific epoch:
+   - Ensures consistency between indexed data and actual chain state
+   - Reports any errors found during validation
```

Comment:
Newbie question: is this different than "detecting holes" as discussed below?

Reply:
@BigLep Yes, index corruption could also be:

1) We only have reverted entries for an epoch but no non-reverted entries.
2) We have more than one non-reverted tipset for a given epoch.
3) The number of indexed events for a give tipset does NOT match what's in the chain

and so on and so forth. Basically, "holes" just means missing entries but we could have index corruption other than "holes" too.